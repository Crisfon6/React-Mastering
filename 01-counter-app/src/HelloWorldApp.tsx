export function HelloWorldApp({name}:{name:string}){
    return (
        <>
        <h1>Hello {name}</h1>
        <p>This is a simple React component.</p>
        </>
    );
}