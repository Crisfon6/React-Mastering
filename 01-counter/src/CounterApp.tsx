import { useState } from "react";

export function CounterApp({ value }: { value: number }) {
  const [counter, setCounter] = useState(value);
/*   const handleClick = () => setCounter(counter + 1); */
  const handleClick = (val:number) => val === 0 ? setCounter(value) : setCounter((v)=> v+val);
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={()=>handleClick(1)} name='+1'> +1</button>
      <button onClick={()=>handleClick(-1)} name='-1'> -1</button>
      <button onClick={()=>handleClick(0)} name='reset'>Reset</button>
    </>
  );
}
