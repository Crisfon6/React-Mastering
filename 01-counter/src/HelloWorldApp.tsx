export function HelloWorldApp({ name , subtitle= 'This is a simple React component.'}: { name: string , subtitle?: string }) {
    return (
        <>
            <h1 data-testid="test-title">Hello {name}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
        </>
    );
}