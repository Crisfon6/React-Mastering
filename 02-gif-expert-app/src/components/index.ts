export * from "./AddCategory";
export * from "./CardCategory";
export * from "./GridCategory";
