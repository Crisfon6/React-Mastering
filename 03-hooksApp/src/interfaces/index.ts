export * from './PokeApi.interface';
export * from './RickMortyApi.interface';
export * from './Todo.interface';
