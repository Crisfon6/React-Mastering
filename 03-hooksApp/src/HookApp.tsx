import { CounterApp } from "./01-useState/CounterApp"
import { CounterWithHook } from "./01-useState/CounterWithHook"

export const HookApp = () => {
  return (
    <div className="container">
      <h1 className="text-3xl">
    Hello world!
  </h1>
  <CounterApp />
  <CounterWithHook />
    </div>
  )
}