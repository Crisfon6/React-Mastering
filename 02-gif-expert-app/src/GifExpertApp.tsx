import { AddCategory, GridCategory } from "./components";
import { useState } from "react";

export function GifExpertApp() {
  const [categories, setCategories] = useState(["one punch"]);
  const addCategory = (newCategory: string) => {
    newCategory = newCategory.toLowerCase();
    if (categories.includes(newCategory)) return;
    setCategories((cat: string[]) => [newCategory, ...cat]);
  };
  return (
    <>
      <div className="bg-[url('/bg.gif')] bg-contain bg-center ">
        <h1 className="pt-5 text-4xl md:text-5xl lg:text-6xl font-bold mb-6 text-center text-rose-50 tracking-wide [text-shadow:_-1px_-1px_0_#a855f7,_1px_-1px_0_#a855f7,_-1px_1px_0_#a855f7,_1px_1px_0_#a855f7] py-1 px-1 transform hover:scale-105 transition-transform duration-300">
          GiftSearch
        </h1>
        <div className="flex justify-center container mx-auto px-10">
          <AddCategory
            /* setCategories={setCategories} categories={categories} */
            onNewCategory={addCategory}
          />
        </div>
      </div>
      {categories.map((category) => (
        <GridCategory key={category} category={category} />
      ))}
      <div className="flex justify-center container mx-auto px-10"></div>
    </>
  );
}
