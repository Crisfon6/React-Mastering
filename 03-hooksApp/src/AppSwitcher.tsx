import { useState } from 'react';
import { HookApp } from './HookApp';
import { TodoApp } from './10-useReducer/TodoApp';

export const AppSwitcher = () => {
  const [currentApp, setCurrentApp] = useState<'hook' | 'todo'>('hook');

  return (
    <div className="min-h-screen bg-gray-100">
      <div className="container mx-auto px-4 py-8">
        <div className="flex justify-center gap-4 mb-8">
          <button
            onClick={() => setCurrentApp('hook')}
            className={`px-6 py-3 rounded-lg font-semibold transition-all duration-200 ${
              currentApp === 'hook'
                ? 'bg-blue-600 text-white'
                : 'bg-white text-blue-600 hover:bg-blue-50'
            }`}
          >
            Hook App
          </button>
          <button
            onClick={() => setCurrentApp('todo')}
            className={`px-6 py-3 rounded-lg font-semibold transition-all duration-200 ${
              currentApp === 'todo'
                ? 'bg-blue-600 text-white'
                : 'bg-white text-blue-600 hover:bg-blue-50'
            }`}
          >
            Todo App
          </button>
        </div>

        <div className="bg-white rounded-xl shadow-lg p-6">
          {currentApp === 'hook' ? <HookApp /> : <TodoApp />}
        </div>
      </div>
    </div>
  );
}; 