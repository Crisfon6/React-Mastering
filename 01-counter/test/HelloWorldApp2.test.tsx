import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { HelloWorldApp } from "../src/HelloWorldApp";

describe("HelloWorldApp Component", () => {
  const testName = "Cristian";
  const testSubtitle = "This is a simple React component.";
  test("should have the title in a h1 tag", () => {
    const { container } = render(
      <HelloWorldApp name={testName} subtitle={testSubtitle} />
    );
    expect(container).toMatchSnapshot();
  });
  test("should be the title Hello Cristian", () => {
    render(<HelloWorldApp name={testName} subtitle={testSubtitle} />);
    expect(screen.getByText(`Hello ${testName}`)).toBeInTheDocument();
  });
  test('should have the title in h1 tag',()=>{
    render(<HelloWorldApp name={testName} subtitle={testSubtitle} />);
    expect(screen.getByRole('heading',{level:1})).toHaveTextContent(`Hello ${testName}`);
  })
  test('should have the subtitle in a p tag',()=>{
    render(<HelloWorldApp name={testName} subtitle={testSubtitle} />);
    expect(screen.getAllByText(testSubtitle).length).toBe(2);
  })
});
