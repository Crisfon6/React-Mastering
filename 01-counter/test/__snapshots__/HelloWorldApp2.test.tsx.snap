// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldApp Component should have the title in a h1 tag 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello 
    Cristian
  </h1>
  <p>
    This is a simple React component.
  </p>
  <p>
    This is a simple React component.
  </p>
</div>
`;
